test/transaction/payload/proregtxpayload.js

```diff
@@ -0,0 +1,148 @@
+var expect = require('chai').expect;
+
+var DashcoreLib = require('../../..');
+
+var BufferUtil = DashcoreLib.util.buffer;
+var Payload = DashcoreLib.Transaction.Payload;
+var ProRegTxPayload = Payload.ProRegTxPayload;
+
+var options = {
+  type: 1,
+  mode: 2,
+  collateralIndex: 1,
+  ipAddress: '0000:0000:0000:0000:0000:ffff:c0a8:0001', //192.168.0.1
```

Comment:
according to DIP 3 only IPv4 mapped addresses are allowed. This one here is normal IPv6. IPv4 would be something like
0000:0000:0000:0000:0000:FFFF:192.168.0.1
http://www.ipuptime.net/ipv4mapped.aspx
Tests fail when putting IPv4 mapped address. Also I think you need to make sure about the network byte order when parsing the payload (ipAddress and port)
Why not use Sam's testvectors for testing?

Reply:
I guess you're right its just representational detail. Either way I got you a testvector, which is indeed missing from Sam's collection: 

`0300010003a966c289061610196f43ec4f95192d62864ede64fa4ce5d5c905e1a9200855200000000049483045022100927551e57f3a20e73187b0dd60ca44c0f86e9c63e365b24d2c4dfa79d13075a80220475a615791b52592bfb7165ab1319897aea44d779ea0dd84f35de1336663137401fefffffff64c5645b70855061b477f45d0816564b168fb5dcf038cb83cf085e7e88aa8390000000048473044022074a7afff9ed3bf04ed3f99fa5d59cce7f5e9c4332a3ade32ead195ee0784e54e02202fdf6af49a638b0b0a59cbcad1ea049b67cda22e758b8cdfafadeaec8f2077f401feffffffba08bfaff3611d7b2d97c50cd40cf2967dc7c22073addba8e91215b494cc514e0000000048473044022065048cfe9e4d9ffbdc32a40b029d46e8a221c911a8228ea157ddddf5689cab32022040e4985fd0fb9aedb4b5ab6ec255ea4026e8098e6f74944c03344bfbbf7d908201feffffff02857f357a0a0000001976a914e4d5a109661155a616f05b1a0b2c95221ca0c23388ac00e87648170000001976a91496dc3875f032c9439c4b1e4c3c9ddbd8a9c9594888ac00000000d60100431201000100000000000000000000000000ffff9f596eb84e1f8ce053db1d2dca51baa435e4dbf104dc1c0bd9038ce053db1d2dca51baa435e4dbf104dc1c0bd9038ce053db1d2dca51baa435e4dbf104dc1c0bd9031976a91496dc3875f032c9439c4b1e4c3c9ddbd8a9c9594888ac000017720491c12b97c956257e6865eca1196956eda35fe3dd42e6949019a42a410b411ba498c1a8781be5dab5e9afec7d0b69cb78608ddc93cf8441d3d274b64bce091936f48a8d779ac3e8da6828486f7df21c3c12b7e1ef84b1de4172001ce47fa5de`

It's the hex data in the response for `gettransaction 75aa128db4cd7679fd88206bd6ef71f57e1b6fe04c2da5515193a6fcd40a47eb` in dapinet and gives you the first protxregister made on that devnet. The complete output of that request is

`{
  "amount": 0.00000000,
  "fee": -0.00000635,
  "confirmations": 24861,
  "instantlock": false,
  "blockhash": "000022d003cbff3a747640b67c63debf9110fe67bb331d6a19f1fbf09da5dd5a",
  "blockindex": 3,
  "blocktime": 1535602508,
  "txid": "75aa128db4cd7679fd88206bd6ef71f57e1b6fe04c2da5515193a6fcd40a47eb",
  "walletconflicts": [
  ],
  "time": 1535602436,
  "timereceived": 1535602436,
  "bip125-replaceable": "no",
  "details": [
    {
      "account": "",
      "address": "ya582Any2PqK1r9Juu29ifg2U2c3vx3mcE",
      "category": "send",
      "amount": -1000.00000000,
      "label": "MN1",
      "vout": 1,
      "fee": -0.00000635,
      "abandoned": false
    }, 
    {
      "account": "MN1",
      "address": "ya582Any2PqK1r9Juu29ifg2U2c3vx3mcE",
      "category": "receive",
      "amount": 1000.00000000,
      "label": "MN1",
      "vout": 1
    }
  ],
  "hex": "0300010003a966c289061610196f43ec4f95192d62864ede64fa4ce5d5c905e1a9200855200000000049483045022100927551e57f3a20e73187b0dd60ca44c0f86e9c63e365b24d2c4dfa79d13075a80220475a615791b52592bfb7165ab1319897aea44d779ea0dd84f35de1336663137401fefffffff64c5645b70855061b477f45d0816564b168fb5dcf038cb83cf085e7e88aa8390000000048473044022074a7afff9ed3bf04ed3f99fa5d59cce7f5e9c4332a3ade32ead195ee0784e54e02202fdf6af49a638b0b0a59cbcad1ea049b67cda22e758b8cdfafadeaec8f2077f401feffffffba08bfaff3611d7b2d97c50cd40cf2967dc7c22073addba8e91215b494cc514e0000000048473044022065048cfe9e4d9ffbdc32a40b029d46e8a221c911a8228ea157ddddf5689cab32022040e4985fd0fb9aedb4b5ab6ec255ea4026e8098e6f74944c03344bfbbf7d908201feffffff02857f357a0a0000001976a914e4d5a109661155a616f05b1a0b2c95221ca0c23388ac00e87648170000001976a91496dc3875f032c9439c4b1e4c3c9ddbd8a9c9594888ac00000000d60100431201000100000000000000000000000000ffff9f596eb84e1f8ce053db1d2dca51baa435e4dbf104dc1c0bd9038ce053db1d2dca51baa435e4dbf104dc1c0bd9038ce053db1d2dca51baa435e4dbf104dc1c0bd9031976a91496dc3875f032c9439c4b1e4c3c9ddbd8a9c9594888ac000017720491c12b97c956257e6865eca1196956eda35fe3dd42e6949019a42a410b411ba498c1a8781be5dab5e9afec7d0b69cb78608ddc93cf8441d3d274b64bce091936f48a8d779ac3e8da6828486f7df21c3c12b7e1ef84b1de4172001ce47fa5de"
}`

Reply:
thanks! will update tests with vectors.

Reply:
Just remember that this is the whole transaction, not only a payload. Just like I did yesterday :grin:

Reply:
hmmm good point thanks.